velox/exec/AggregateWindow.cpp

```diff
@@ -341,7 +348,34 @@ class AggregateWindowFunction : public exec::WindowFunction {
     });
 
     // Set null values for empty (non valid) frames in the output block.
-    setNullEmptyFramesResults(validRows, resultOffset, result);
+    setEmptyFramesResult(validRows, resultOffset, result, emptyResult_);
+  }
+
+  // Precompute and save the aggregate output for empty input in emptyResult_.
+  // This value is returned for rows with empty frames.
+  void computeDefaultAggregateValue(
+      const std::string& name,
+      const TypePtr& resultType) {
+    constexpr int kRowSizeOffset = 8;
+    constexpr int kOffset = kRowSizeOffset + 8;
+    aggregate_->setOffsets(kOffset, 0, 1, kRowSizeOffset);
+    std::vector<char> group(kOffset + aggregate_->accumulatorFixedWidthSize());
+    std::vector<char*> groups(1, group.data());
+    aggregate_->initializeNewGroups(
+        groups.data(), std::vector<vector_size_t>{0});
+
+    // addSingleGroupRawInput is called to catch the expected errors for
+    // incorrect inputs with the aggregate fuzzer.
+    char dummyGroup{
+        static_cast<char>(kOffset + aggregate_->accumulatorFixedWidthSize())};
+    SelectivityVector dummyRows(1, false);
+    aggregate_->addSingleGroupRawInput(
+        &dummyGroup, dummyRows, argVectors_, false);
+
+    emptyResult_ = BaseVector::create(resultType, 1, pool_);
```

Comment:
Can we make this a constantVector ? That way you needn't call copy for each index at a time.

Reply:
Is this regarding the copy in `setEmptyFramesResult` ? Could you please share how this could be avoided for each index with a `ConstantVector`?

Reply:
I think your current logic is fine now.